### DISCUSSION TITLE
Blur some area

### DISCUSSION THREAD

**Discussion by JA16122000** [mrlt8/docker-wyze-bridge#919] [Ideas]
Hi!

Is it possible, to blur some area ?
I would love this feature ;)

**Comment by mrlt8** [comment]
Possibly with a custom ffmpeg command. Would definitely require re-encoding.

**Reply by JA16122000** [reply]
Something like this i asume?
https://superuser.com/questions/901099/ffmpeg-apply-blur-over-face

I will look in to this.

Is this possible to add as an option ?

**Reply by mrlt8** [reply]
Could you try the dev branch? I added an option to use custom video filters.

You should be able to use something like this (where `front` is the name of the cam):

```yaml
- FFMPEG_FILTER_FRONT=drawbox=x=1300:y=1000:w=600:h=55:color=black:t=fill
```
or for all cameras:
```yaml
- FFMPEG_FILTER=drawbox=x=1300:y=1000:w=600:h=55:color=black:t=fill
```
This example will draw a black box around the timestamp.

**Reply by JA16122000** [reply]
I'm on vacation for a few days. 
As soon as I get home (in 4 days) I can try it out!


I'm now using:

MTX_PATHS_ACHTERTUIN_RUNONREADY=ffmpeg -i rtsp://blablabla.nl:8554/achtertuin -c copy -f segment -strftime 1 -segment_time 3600 -segment_format mpegts /record//achtertuin/Achtertuin_%Y-%m-%d_%H-%M-%S.mkv

So i record the streams.
I can leave that staying active?

**Reply by JA16122000** [reply]
Back from a short brake and testing:

![Schermafbeelding 2023-07-20 152503](https://github.com/mrlt8/docker-wyze-bridge/assets/112784196/34f17e4d-675c-41a8-ad97-a661c7681f84)
Looks fine!

Now its black, is it also possible to do it like this? 

![blur-house-on-apple-maps 1](https://github.com/mrlt8/docker-wyze-bridge/assets/112784196/da58c3ba-84ff-4a33-833f-901148d0476b)

**Reply by mrlt8** [reply]
Yes, but it could potentially require more processing power? You'll have to play around with that and see what works best for you. Depending on your hardware, you may even be able to utilize some hardware accelerated encoding with certain filters.

**Reply by JA16122000** [reply]
I would love to try that sort kind of blurring.
What kind of filter do i have to use for the example i showed?
Is that allreadey possible for us and do you have an example ENV ?

**Reply by JA16122000** [reply]
Can you provide us some examples @mrlt8 ? :D

**Reply by mrlt8** [reply]
Added `FFMPEG_FILTER_COMPLEX` to the dev branch. However, as this is an extremely niche feature and custom commands are already supported, I can't offer much support for this right now.

**Reply by JA16122000** [reply]
I used the "dev" branche:

> FFMPEG_COMPLEX_VOORTUIN [0:v]crop=400:400:60:180,avgblur=10[fg];[0:v][fg]overlay=60:180[v]

Wel, it is working!! but as you mentioned, a lot of power is needed to do this!
